### ISSUE TITLE
Subsetting: WASHb Kenya histogram for "Time to water source" is not showing all of the data

### ISSUE THREAD

**Issue by danicahelb** [VEuPathDB/web-eda#515]
In the WASHb Kenya study, the histogram on the subset tab for 'Time to water source' has an x-axis that extends from 600, but the data seems to stop ~200 (even after changing to log scale & zooming in). The range indicator states that the data ranges from 0-600. 

![image](https://user-images.githubusercontent.com/25450900/135535556-7b4ae69d-adce-4920-bd45-18481ef15b09.png)

Using the visualization histogram, I can see the data at ~600.
![image](https://user-images.githubusercontent.com/25450900/135535613-96453ee6-774a-4135-a9e0-005f056e7d4c.png)

Live site data also extends to ~600.
![image](https://user-images.githubusercontent.com/25450900/135535671-2699a6bd-c466-4a34-9bf7-013a3548458f.png)

**Comment by dmfalke** [comment]
After giving this some thought, and trying _really_ hard to find a way to send it back to EdaSubsettingService, I agree that it belongs here. In this particular case, the client should set the `displayRangeMax` to `rangeMax + binSize`. Should we do the same for an annotated `displayRangeMax`?

**Comment by dmfalke** [comment]
However... I do think the viz and distribution service should handle inclusive/exclusive the same for bins. But that can probably be handled separately.

**Comment by d-callan** [comment]
Yes. We should I think do this for the annotated display range too, because the person who sets that value has no way of knowing what bin width the user will choose. And if the viz tab is doing anything differently I think it's a bug, or some default behavior I failed to override.

**Comment by dmfalke** [comment]
For bins:
* in subsetting: `[min, max)`
* in viz: `(min, max]` (the first bin is `[min, max]`)

**Comment by bobular** [comment]
Hi guys.

I agree this is a front-end issue. There was a time when we calculated the axis range from max(binEnd) and then we didn't, for some reason!  I'll claim this and fix it.

**Comment by bobular** [comment]
> For bins:
> 
>     * in subsetting: `[min, max)`
> 
>     * in viz: `(min, max]` (the first bin is `[min, max]`)

Oh dear - see https://github.com/VEuPathDB/EdaNewIssues/issues/38 !!

**Comment by bobular** [comment]
> I agree this is a front-end issue. There was a time when we calculated the axis range from max(binEnd) and then we didn't, for some reason! I've usually tried hard to make sure the histogram component doesn't omit data it has been given, but again, this must have slipped somehow.

Actually the code is still there that uses something like max(binEnd) - so I'll have to investigate further why that last bin isn't showing.

**Comment by dmfalke** [comment]
@bobular in this particular case, the data range is `0 - 600`, and the bin size is `25`. It happens that `25` divides evenly into `600`, so the last bin has `600` as its max. Since the distribution endpoint used by subsetting treats bins as max-exclusive, the elements associated with `600` are not included. However, in the visualization plugin, bins are max-inclusive, and so include the elements associates with `600`. I would bet that if bin size were not a divisor of `600`, we would get all data.

I think the current proposal to add the bin width to rangeMax or displayRangeMax makes the most sense.

**Comment by ryanrdoherty** [comment]
@dmfalke With the new code deployed today, the final bin returned by subsetting distribution is now guaranteed to contain the data range max (even if it means creating a "remainder bin" that ONLY contains values exactly matching the max).  Given this, can you simply use the first bin min and last bin max as the view port?

**Comment by dmfalke** [comment]
Hmm... users can specify the viewport, so I'm a little hesitant to use something different than what was requested. I'm more inclined to extend the max of the viewport for the default. Otherwise, we have to determine if the range was user-supplied or algorithmically derived.

**Comment by bobular** [comment]
I'll figure something out, don't worry guys!

**Comment by bobular** [comment]
There you go.
It was in "next cycle" but what the hell!

**Comment by danicahelb** [comment]
looks great, thanks team!
